tests/mock_network/mod.rs

```diff
@@ -443,11 +443,11 @@ fn check_section_info_ack() {
     let nodes = create_connected_nodes_until_split(&network, vec![1, 1], true);
 
     // Due to the behaviour in the `create nodes until split`, there shall only be one section nodes
-    // sending out `section_info_ack`. Hence only one section shall receive the ack.
-    let num_of_receives = nodes
+    // sending out `section_info_ack`. Hence only one section shall consensus on it eventually.
+    let num_of_consensus = nodes
         .iter()
-        .filter(|node| node.inner.received_section_info_ack())
+        .filter(|node| node.inner.get_version() > min_section_size as u64)
         .count();
-    assert!(num_of_receives >= min_section_size);
-    assert!(num_of_receives < 2 * min_section_size);
+    assert!(num_of_consensus >= min_section_size);
```

Comment:
I am ok with this assert in this PR, but I am wondering if this reflects a problem that we will have to solve in a follow up PR (i.e shouldn't both side know the other their_knowledge)?

Reply:
Yeah, I agree. As the function used to setup is `adding nodes until split`. So at the tip of split, only the section that not containing the last node (which causes the split), will have

Reply:
Strange, when the last node is added, it change both section since they are both the result of the split (But maybe a message is not sent, or consensus not reached somewhere).